app/src/main/java/net/justdave/mcstatus/MinecraftServer.java

```diff
@@ -324,21 +324,28 @@ public void query() {
 		}
 	}
 
-	private int readVarInt(InputStream in) throws InputMismatchException {
+	private int readVarInt(InputStream in) {
 		int theInt = 0;
-		int byteCounter = 0;
+//		int byteCounter = 0;
 
-		while (true) {
+		for (int i = 0; i < 6; i++){
```

Comment:
Just read 5 bytes instead of throwing when we get more. However, there is a check for the 'invalid' byte I receive. Maybe more checks should be added, but this works for my use case.